src/create-action.ts

```diff
@@ -2,13 +2,31 @@ import { StringType, ActionCreator } from './types';
 import { validateActionType } from './utils/utils';
 import { action } from './action';
 
-export type PayloadMetaAction<T extends StringType, P, M> = P extends undefined
-  ? M extends undefined
-    ? { type: T }
-    : { type: T; meta: M }
-  : M extends undefined
-  ? { type: T; payload: P }
-  : { type: T; payload: P; meta: M };
+type ActionWithType<T> = { type: T };
+type ActionWithPayload<T, P> = { type: T; payload: P };
+type ActionWithMeta<T, M> = { type: T; meta: M };
+type ActionWithPayloadAndMeta<T, P, M> = { type: T; payload: P; meta: M };
+type ActionWithError<T, E> = { type: T; error: E };
```

Comment:
ActionWithError type is the same as ActionWithMeta, without constrains it will not work